### PR TITLE
Adding is process_group initialized support

### PR DESCRIPTION
**Pull request by teng-li**
Added this check as a utility function so that the user can avoid to initialize it twice and get an error.  Also did minor bug fix for destroy process group function

### CODE REVIEW

File: torch/distributed/__init__.py [hunk 1]
```diff
@@ -37,8 +37,17 @@ def destroy_process_group():
...
     """
     global _backend
+    global _initialized
+    torch._C._dist_destroy_process_group()
     _backend = dist_backend.UNDEFINED
-    return torch._C._dist_destroy_process_group()
+    _initialized = 0
+
+
+def is_process_group_initialized():
```
Comment (apaszke) [inline torch/distributed/__init__.py]: I think `is_initialized` would be enough

File: torch/distributed/__init__.py [hunk 2]
```diff
@@ -37,8 +37,17 @@ def destroy_process_group():
...
+    """Checking if the process group has been initialized
+    """
+    global _initialized
```
Comment (apaszke) [inline torch/distributed/__init__.py]: this is unnecessary. you're never writing to it

File: torch/distributed/__init__.py [hunk 3]
```diff
@@ -37,8 +37,17 @@ def destroy_process_group():
     Destroy the initialized distributed package
     """
     global _backend
+    global _initialized
+    torch._C._dist_destroy_process_group()
     _backend = dist_backend.UNDEFINED
-    return torch._C._dist_destroy_process_group()
+    _initialized = 0
```
Comment (apaszke) [inline torch/distributed/__init__.py]: do we even need this flag? We could treat `_backend is `dist_backend.UNDEFINED`  as `not _initialized`

Comment (teng-li) [reply torch/distributed/__init__.py]: _Initialized has either the value of _INITIALIZED_PG or _INITIALIZED_MW and has been used in a lot of the places in the file. Let's keep it for now.  Removed unnecessary global and rename the function.

**Review by apaszke — APPROVED** [review]

**Pull request merged**